connectors/sources/mssql.py

```diff
@@ -62,6 +62,244 @@ def all_schemas(self):
+                    schema=self.schema,
+                    table=table,
+                ),
+            )
+        )
+
+    async def get_table_last_update_time(self, table):
+        return await anext(
+            self._execute_query(
+                query=self.queries.table_last_update_time(
+                    schema=self.schema,
+                    table=table,
+                ),
+            )
+        )
+
+    async def data_streamer(self, table):
+        """Streaming data from a table
+
+        Args:
+            table (str): Table.
+
+        Raises:
+            exception: Raise an exception after retrieving
+
+        Yields:
+            list: It will first yield the column names, then data in each row
+        """
+        async for data in self._execute_query(
+            query=self.queries.table_data(
+                schema=self.schema,
+                table=table,
+            ),
+            fetch_many=True,
+            schema=self.schema,
+            table=table,
+        ):
+            yield data
+
+    async def _execute_query(self, query, fetch_many=False, **kwargs):
+        """Executes a query and yield rows
+
+        Args:
+            query (str): Query.
+            fetch_many (bool): Flag to use fetchmany method. Defaults to False.
+
+        Raises:
+            exception: Raise an exception after retrieving
+
+        Yields:
+            list: Column names and query response
+        """
+        retry = 1
+        yield_once = True
+
+        rows_fetched = 0
+
+        while retry <= self.retry_count:
+            try:
+                cursor = await self.get_cursor(query=query)
+                if fetch_many:
+                    # sending back column names only once
+                    if yield_once:
+                        yield [
+                            f"{kwargs['schema']}_{kwargs['table']}_{column}".lower()
+                            for column in cursor.keys()  # pyright: ignore
+                        ]
+                        yield_once = False
+
+                    while True:
+                        rows = cursor.fetchmany(size=self.fetch_size)  # pyright: ignore
+                        rows_length = len(rows)
+
+                        if not rows_length:
+                            break
+
+                        for row in rows:
+                            yield row
+
+                        rows_fetched += rows_length
+                        await asyncio.sleep(0)
+                else:
+                    yield cursor.fetchall()  # pyright: ignore
+                break
+            except (InternalClientError, ProgrammingError):
+                raise
+            except Exception as exception:
+                self._logger.warning(
+                    f"Retry count: {retry} out of {self.retry_count}. Exception: {exception}"
+                )
+                if retry == self.retry_count:
+                    raise exception
```

Comment:
```suggestion
                    raise
```

Reply:
This is not needed once #1401 is merged.